components/micro-gateway-core/src/main/ballerina/src/gateway/filters/subscription_filter.bal

```diff
@@ -43,137 +44,155 @@ public type SubscriptionFilter object {
-                    authenticationContext.callerToken = jwtToken;
-                    authenticationContext.applicationId = decodedPayload.application.id.toString();
-                    authenticationContext.applicationName = decodedPayload.application.name.toString
-                    ();
-                    authenticationContext.applicationTier = decodedPayload.application.tier.toString
-                    ();
+                    if (decodedPayload.application.id != null) {
+                        authenticationContext.applicationId = decodedPayload.application.id.toString();
+                    }
+                    if (decodedPayload.application.name != null) {
+                        authenticationContext.applicationName = decodedPayload.application.name.toString
+                        ();
+                    }
+                    if (decodedPayload.application.tier != null) {
+                        authenticationContext.applicationTier = decodedPayload.application.tier.toString
+                        ();
+                    }
                     authenticationContext.subscriber = decodedPayload.application.owner.toString();
                     authenticationContext.consumerKey = decodedPayload.consumerKey.toString();
-                    authenticationContext.apiTier = subscription.subscriptionTier.toString();
-                    authenticationContext.apiPublisher = subscription.publisher.toString();
-                    authenticationContext.subscriberTenantDomain = subscription
-                    .subscriberTenantDomain.toString();
                     authenticationContext.keyType = decodedPayload.keytype.toString();
-                    // setting keytype to invocationContext
-                    printDebug(KEY_SUBSCRIPTION_FILTER, "Setting key type as " +
-                            authenticationContext.keyType);
-                    runtime:getInvocationContext().attributes[KEY_TYPE_ATTR] = authenticationContext
-                    .keyType;
+                    runtime:getInvocationContext().attributes[KEY_TYPE_ATTR] = authenticationContext.
+                    keyType;
                     filterContext.attributes[AUTHENTICATION_CONTEXT] = authenticationContext;
-                    printDebug(KEY_SUBSCRIPTION_FILTER, "Subscription validation success.");
                     return true;
                 }
-                index = index + 1;
+                int index = 0;
+                while (index < l) {
+                    var subscription = subscribedAPIList[index];
+                    string apiName="";
+                    string apiVersion="";
+                    if(apiConfig is APIConfiguration){
+                        apiName= apiConfig.name;
+                        if(apiConfig?.apiVersion is string){
+                            apiVersion = apiConfig?.apiVersion;
+                        }
+                    }
+                    if (subscription.name.toString() == apiName &&
```

Comment:
How about continuing the loop when invert condition is met? That way we can avoid the long if block